blocks/ifus/ifus.js

```diff
@@ -0,0 +1,248 @@
+
+  countryProducer().forEach((country) => {
+    createDomStructure([
+      { type: 'option', attributes: { value: country.ISO_3166_1_alpha_2_code }, textContent: country['Country Name'] },
+    ], countrySelector);
+  });
+  countrySelector.disabled = false;
+
+  searchButton.disabled = false;
+  searchButton.addEventListener('click', async () => {
+    clear.forEach((value) => { value.options[0].selected = true; });
+    const assets = idSelector.value && countrySelector.value
+      ? assetFilter(idSelector, countrySelector) : [];
+    result.innerHTML = '';
+    createDomStructure([{ type: 'h4', textContent: await translate('ifuSearchTitle', 'Search results') }], result);
+    if (assets.length === 0) {
+      createDomStructure([{ type: 'div', children: [{ type: 'strong', textContent: await translate('ifuSearchNoResult', 'No result') }] }], result);
+    } else {
+      const map = new Map();
+      assets.forEach((asset) => {
+        if (map.has(asset.Title)) {
+          const links = map.get(asset.Title);
+          links.push(asset.URL);
+        } else {
+          map.set(asset.Title, [asset.URL]);
+        }
+      });
+      await Promise.all(Array.from(map).map(async ([key, value]) => {
+        createDomStructure([
+          {
+            type: 'div',
+            children: [
+              { type: 'h5', textContent: key },
+              {
+                type: 'div',
+                children: [
+                  {
+                    type: 'div',
+                    children:
+                      [
+                        {
+                          type: 'div',
+                          textContent: productCodes(idSelector),
+                          children:
+                            [
+                              { type: 'h6', position: 'prepend', textContent: await translate('ifuSearchProductCodes', 'Product Code(s)') },
+                            ],
+                        },
+                      ],
+                  },
+                  {
+                    type: 'div',
+                    textContent: countrySelector.value,
+                    children:
+                      [
+                        { type: 'h6', position: 'prepend', textContent: await translate('ifuSearchCountrySelected', 'Country Selected') },
+                      ],
+                  },
+                  {
+                    type: 'div',
+                    children: value.map(
+                      (link) => ({
+                        type: 'a', attributes: { href: link, target: 'blank' }, children: [{ type: 'button', textContent: link.substring(link.lastIndexOf('/') + 1) }],
+                      }),
+                    ),
+                  },
+                ],
+              }],
+          },
+        ], result);
+      }));
+    }
+    result.classList.remove('no-result');
+  });
+}
+
+async function populate(block) {
+  const resp = await fetch('/en/dj-test.json?limit=10000');
```

Comment:
for readability and less indentation i would use a guard clause like this:

```
  const resp = await fetch('/en/dj-test.json?limit=10000');
  if (!resp.ok) {
    throw new Error(`${resp.status}: ${resp.statusText}`);
  }

  const json = await resp.json();
```

Reply:
Done.